### PR TITLE
Implement OP_CHECKDATASIG{VERIFY}

### PR DESCRIPTION
**Pull request by tuxcanfly**
Implemented OP_CHECKDATASIG, OP_CHECKDATASIGVERIFY

Refs: https://reviews.bitcoinabc.org/rABC9ba4bfca513d6386ee3d313b15bdd4584da7633d

### CODE REVIEW

File: lib/script/script.js [hunk 1]
```diff
@@ -1131,7 +1131,7 @@ class Script {
             || !(sig[sig.length - 1] & Script.hashType.SIGHASH_FORKID))
             subscript.findAndDelete(sig);
 
-          validateSignature(sig, flags);
+          validateTxSignature(sig, flags);
```
Comment (nodar-chkuaselidze) [inline lib/script/script.js]: validateTXSignature

Comment (nodar-chkuaselidze) [inline lib/script/script.js]: validateTXSignature

File: test/magnetic-test.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  common.flags.MANDATORY_VERIFY_FLAGS
+];
+
+// TODO: update standard flags
+flags = flags.map(flag => flag & ~common.flags.VERIFY_COMPRESSED_PUBKEYTYPE);
+
+const priv = '0000000000000000000000000000000000000000000000000000000000000001';
+const key = Keyring.fromPrivate(Buffer.from(priv, 'hex'));
+
+function isSuccess(stack, script, expected) {
```
Comment (nodar-chkuaselidze) [inline test/magnetic-test.js]: Add `flags` as param instead of global var.

Comment (tuxcanfly) [reply test/magnetic-test.js]: Let's do this after release :+1:

**Pull request merged**